bundle/terranova/tnresources/adapter.go

```diff
@@ -279,21 +365,46 @@ func (a *Adapter) DoDelete(ctx context.Context, id string) error {
 	return nil
 }
 
-func (a *Adapter) DoCreate(ctx context.Context, config any) (string, error) {
+func (a *Adapter) DoCreate(ctx context.Context, config any) (string, any, error) {
+	if a.doCreate == nil {
+		return "", nil, errors.New("internal error: DoCreate not found")
+	}
+
 	outs, err := a.doCreate.Call(ctx, config)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	// no error checking, type is enforced via calladapt + interface
 	id := outs[0].(string)
-	return id, nil
+
+	if len(outs) == 2 {
```

Comment:
Can we have some helper function like `isWithRefreshVariant(caller *calladapt.BoundCaller, expectedOutCount int)` whcih will check this? make it a bit more readable

Reply:
we don't store this info on BoundCaller; added a comment to make it more clear 1474b3ad9fa6d864fb84c641ca9e1edf5844c09a